src/main/java/bookstore/mapper/UserMapper.java

```diff
@@ -0,0 +1,23 @@
+package bookstore.mapper;
+
+import bookstore.dto.RegisterRequestDto;
+import bookstore.dto.RegisterResponseDto;
+import bookstore.entity.role.Role;
+import bookstore.entity.user.User;
+import java.util.HashSet;
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.Mapper;
+import org.mapstruct.NullValueCheckStrategy;
+import org.mapstruct.ReportingPolicy;
+
+@Mapper(componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
+public interface UserMapper {
+    RegisterResponseDto toRegisterResponseDto(User user, String token);
+
+    User toModel(RegisterRequestDto userRequestDto, HashSet<Role> roles);
```

Comment:
It seems a bit strange to me because we already know that we're using a mapper for users because of the naming UserMapper.toModel. And the "UserMapper.toUser" is like a tautology. It could be a more common style to use all mappers in this name convention.